### PR TITLE
Port credential_process support from Java SDK V1

### LINKED ISSUE #455: credential_process is not implemented
The SDK is not compatible with an external `credential_process`

## Expected Behavior
Specifying a `credential_process` in the credentials file should allow a user of the sdk to authenticate.

## Current Behavior
The `credential_process` option is ignored.

## Possible Solution
I have some thoughts about how to implement this and I would be happy to work on a PR, I just wanted to make sure that was something that would be useful. If so we can talk more about an implementation.

## Steps to Reproduce
- Put a `credentials_process` line in `~/.aws/credentials`
- Try to build a client with a `ProfileCredentialsProvider`

## Your Environment
* AWS Java SDK version used: 2.0.0-preview-9
* JDK version used: 1.8.0_161

### PR DESCRIPTION
**Pull request by abrooksv**
## Description
Lift-and-shift support from V1 and switch to using V2 utils and conventions

## Motivation and Context
To unblock AWS Toolkit for JetBrains customers

Fixes #455

## Testing
Need to do more testing, but feel free to review.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java [hunk 1]
```diff
@@ -125,6 +130,14 @@ private AwsCredentialsProvider sessionProfileCredentialsProvider() {
         return StaticCredentialsProvider.create(credentials);
     }
 
+    private AwsCredentialsProvider credentialProcessCredentialsProvider() {
+        requireProperties(ProfileProperty.CREDENTIAL_PROCESS);
+
+        return ProcessCredentialsProvider.builder()
+                                  .command(properties.get(ProfileProperty.CREDENTIAL_PROCESS))
+                                  .build();
```
Comment (millems) [inline core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java]: Nit: format

File: core/auth/src/test/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.Platform;
+
+public class ProcessCredentialsProviderTest {
+    private static String scriptLocation;
+ 
+    @BeforeClass
+    public static void setup() throws IOException {
+        String scriptClasspathFilename = Platform.isWindows() ? "windows-credentials-script.bat"
+                                                              : "linux-credentials-script.sh";
```
Comment (millems) [inline core/auth/src/test/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProviderTest.java]: Did you test on windows?

**Comment by abrooksv** [comment]
Addressed feedback, added another test for making sure profile works

File: core/auth/src/test/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProviderTest.java (hunk 2 above)
Comment (abrooksv) [reply core/auth/src/test/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProviderTest.java]: Ran `mvn clean install -pl :auth`

**Comment by millems** [comment]
Can you rebase? Can you confirm that this pull request can be released under the Apache 2 license?

**Comment by abrooksv** [comment]
> Can you rebase? Can you confirm that this pull request can be released under the Apache 2 license?

Since I work for AWS, yes?

**Comment by zoewangg** [comment]
@abrooksv You left the license box unchecked! :trollface:

**Pull request merged**